tests/preprocessing/test_trips.py

```diff
@@ -0,0 +1,87 @@
+import os
+import pandas as pd
+import pytest
+
+import trackintel as ti
+
+
+class TestGenerate_tours:
+    """Tests for generate_tours() method."""
+
+    def test_generate_trips(self):
+        trips = ti.io.file.read_trips_csv(os.path.join("tests", "data", "geolife_long", "trips.csv"), index_col="id")
+        trips_, tours = ti.preprocessing.trips.generate_tours(trips, max_dist=100)
+
+        tours_loaded = ti.io.file.read_tours_csv(
+            os.path.join("tests", "data", "geolife_long", "tours.csv"), index_col="id"
+        )
+        # check only first five columns because journey and location ID are empty in this case
+        pd.testing.assert_frame_equal(tours_loaded.iloc[:, :5], tours.iloc[:, :5])
+
+    def test_tours_location(self):
+        """Test generate tours method with the locations given as an argument"""
+        # create trips from geolife (based on positionfixes)
+        pfs, _ = ti.io.dataset_reader.read_geolife(os.path.join("tests", "data", "geolife_long"))
+        pfs, stps = pfs.as_positionfixes.generate_staypoints(
+            method="sliding", dist_threshold=25, time_threshold=5, gap_threshold=1e6
+        )
+        stps = stps.as_staypoints.create_activity_flag(time_threshold=15)
+        pfs, tpls = pfs.as_positionfixes.generate_triplegs(stps)
+        stps, tpls, trips = ti.preprocessing.triplegs.generate_trips(stps, tpls, gap_threshold=15)
+        # generate locations
+        stps, locs = stps.as_staypoints.generate_locations(method="dbscan", epsilon=100, num_samples=1)
+
+        tours_loaded = ti.io.file.read_tours_csv(
+            os.path.join("tests", "data", "geolife_long", "tours.csv"), index_col="id"
+        )
+        tours_loaded["journey"] = tours_loaded["journey"].astype(object)  # need to cast because of NaNs
+
+        trips_out, tours = ti.preprocessing.trips.generate_tours(trips, stps_w_locs=stps)
+
+        pd.testing.assert_frame_equal(tours_loaded, tours)
+
+    def test_accessor(self):
+        """Test if the accessor leads to the same results as the explicit function."""
+        # create trips from geolife (based on positionfixes)
+        pfs, _ = ti.io.dataset_reader.read_geolife(os.path.join("tests", "data", "geolife_long"))
+        pfs, stps = pfs.as_positionfixes.generate_staypoints(
+            method="sliding", dist_threshold=25, time_threshold=5, gap_threshold=1e6
+        )
+        stps = stps.as_staypoints.create_activity_flag(time_threshold=15)
+        pfs, tpls = pfs.as_positionfixes.generate_triplegs(stps)
+        stps, tpls, trips = ti.preprocessing.triplegs.generate_trips(stps, tpls, gap_threshold=15)
```

Comment:
If you have the same preprocessing/data loading steps several times, you increase readability by creating a [fixture](https://docs.pytest.org/en/6.2.x/fixture.html)